### PR TITLE
Kokoro AWS integration improvements

### PR DESCRIPTION
**Pull request by jtattermusch**
Many smaller improvements (see commits for individual changes).

### CODE REVIEW

**Comment by jtattermusch** [comment]
With the current setup I've been able to get a successful bazel build.
sponge/c88eebb3-e707-4993-80d9-b14a5eed57d3

**Comment by jtattermusch** [comment]
@amidlash please review.

File: tools/internal_ci/linux/grpc_aws_experiment.sh [hunk 1]
```diff
@@ -29,6 +45,12 @@ sudo apt update && sudo apt install -y jq
 AWS_MACHINE_IMAGE=ami-026141f3d5c6d2d0c
 AWS_INSTANCE_TYPE=t4g.xlarge
 AWS_SECURITY_GROUP=sg-021240e886feba750
+# Max allowed lifespan of the AWS instance. After this period of time, the instance will
+# self-terminate (delete itself).
+AWS_INSTANCE_MAX_LIFESPAN_MINS=120
+# increase the size of the root volume so that builds don't run out of disk space
+AWS_STORAGE_SIZE_GB=75
+AWS_DEVICE_MAPPING="DeviceName='/dev/sda1',Ebs={VolumeSize=${AWS_STORAGE_SIZE_GB}}"
```
Comment (amidlash) [inline tools/internal_ci/linux/grpc_aws_experiment.sh:53]: Did this work for you? When I tried sda1 the device wasnt actually created. /dev/sda1 is the normal root of the system.

File: tools/internal_ci/linux/grpc_aws_experiment.sh [hunk 2]
```diff
@@ -66,12 +89,18 @@ echo "Waiting 2m for instance to initialize..."
...
-REMOTE_SCRIPT_FAILURE=0
-echo "Copying to remote instance..."
-scp -i ~/.ssh/temp_client_key -r github/grpc ubuntu@$IP:
+REMOTE_SCRIPT_EXITCODE=0
+echo "Copying workspace to remote instance..."
+# use rsync over ssh since it's much faster than scp
+time rsync -e "ssh -i ~/.ssh/temp_client_key" -a github/grpc ubuntu@$IP:~/workspace
 echo "Beginning CI workload..."
-ssh -i ~/.ssh/temp_client_key ubuntu@$IP "uname -a; ls -l; bash grpc/tools/internal_ci/linux/$WORKLOAD" || REMOTE_SCRIPT_FAILURE=$?
+ssh -i ~/.ssh/temp_client_key ubuntu@$IP "uname -a; cd ~/workspace; ls -l; bash grpc/tools/internal_ci/linux/$WORKLOAD" || REMOTE_SCRIPT_EXITCODE=$?
```
Comment (amidlash) [inline tools/internal_ci/linux/grpc_aws_experiment.sh:97]: We can drop ls -l, which was for debugging only.

File: tools/internal_ci/linux/grpc_aws_experiment_remote.sh [hunk 3]
```diff
@@ -7,7 +22,5 @@ sudo pip install six
 
 cd grpc
 # build with bazel
-export CXX=clang++
-export CC=clang
-tools/bazel test --config=dbg //test/...
+tools/bazel build --config=opt //test/...
```
Comment (amidlash) [inline tools/internal_ci/linux/grpc_aws_experiment_remote.sh:25]: In my experience, bazel fails to find the toolchain unless explicitly set.

File: tools/internal_ci/linux/grpc_aws_experiment.sh (hunk 1 above)
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_aws_experiment.sh:53]: yes it does. It actually changes the default size of the root filesystem (normally 10GB, determined by the image) to whatever you specify here. I ran `df -h` on the workers and that confirmed it. Also stuff that was failing before is now passing.

File: tools/internal_ci/linux/grpc_aws_experiment.sh (hunk 2 above)
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_aws_experiment.sh:97]: sg, we can do in a followup.

File: tools/internal_ci/linux/grpc_aws_experiment_remote.sh (hunk 3 above)
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_aws_experiment_remote.sh:25]: It does work just fine: sponge/c88eebb3-e707-4993-80d9-b14a5eed57d3

**Comment by jtattermusch** [comment]
Since there were no major concern, I'm going to merge this PR to speed thing up (despite not having a formal LGTM yet), but I can address any additional feedback as a followup PR.

**Pull request merged**